subcommand/tls-init/command.go

```diff
@@ -0,0 +1,368 @@
+	if err != nil {
+		c.log.Error("error parsing signer from private key", "err", err)
+		return 1
+	}
+
+	c.log.Info("parsing CA certificate from PEM string")
+	caCert, err := cert.ParseCert([]byte(ca))
+	if err != nil {
+		c.log.Error("error parsing CA certificate from PEM string", "err", err)
+		return 1
+	}
+
+	c.log.Info("generating server certificate and private key")
+	serverCert, serverKey, err := cert.GenerateCert(name, c.getDaysAsDuration(), caCert, signer, hosts)
+	if err != nil {
+		c.log.Error("error generating server certificate and private key", "err", err)
+		return 1
+	}
+
+	serverCertSecret, err := c.clientset.CoreV1().Secrets(c.k8sNamespace).Get(c.ctx, fmt.Sprintf("%s-server-cert", c.namePrefix), metav1.GetOptions{})
+	if err != nil && k8serrors.IsNotFound(err) {
+		c.log.Info("creating server certificate and private key secret")
+		_, err := c.clientset.CoreV1().Secrets(c.k8sNamespace).Create(c.ctx, &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: c.k8sNamespace,
+				Name:      fmt.Sprintf("%s-server-cert", c.namePrefix),
+			},
+			Data: map[string][]byte{
+				corev1.TLSCertKey:       []byte(serverCert),
+				corev1.TLSPrivateKeyKey: []byte(serverKey),
+			},
+			Type: corev1.SecretTypeTLS,
+		}, metav1.CreateOptions{})
+		if err != nil {
+			c.log.Error("error creating server certificate secret in kubernetes", "err", err)
+			return 1
+		}
+	} else if err == nil {
+		serverCertSecret.Data = map[string][]byte{
+			corev1.TLSCertKey:       []byte(serverCert),
+			corev1.TLSPrivateKeyKey: []byte(serverKey),
+		}
+		c.log.Info("updating server certificate and private key secret")
+		_, err := c.clientset.CoreV1().Secrets(c.k8sNamespace).Update(c.ctx, serverCertSecret, metav1.UpdateOptions{})
+		if err != nil {
+			c.log.Error("error updating server certificate secret in kubernetes", "err", err)
+			return 1
+		}
+	} else {
+		c.log.Error("error reading server certificate secret from kubernetes", "err", err)
+		return 1
+	}
+
+	return 0
+}
+
+// getDaysAsDuration returns number of days the certificate
+// is valid for as a time.Duration.
+func (c *Command) getDaysAsDuration() time.Duration {
+	duration, err := time.ParseDuration(fmt.Sprintf("%dh", 24*c.days))
+	if err != nil {
+		c.log.Error("error parsing duration from days", "err", err)
+		return 1
+	}
+	return duration
+}
+
+func (c *Command) init() {
+	c.flags = flag.NewFlagSet("", flag.ContinueOnError)
+	c.flags.IntVar(&c.days, "days", 1825, "Provide number of days the CA is valid for from now on. Defaults to 5 years.")
+	c.flags.StringVar(&c.domain, "domain", "consul", "Domain of consul cluster. Only used in combination with -name-constraint. Defaults to consul.")
+	c.flags.StringVar(&c.caFile, "ca", "", "Provide path to the ca.")
+	c.flags.StringVar(&c.keyFile, "key", "", "Provide path to the key.")
+	c.flags.StringVar(&c.dc, "dc", "dc1", "Provide the datacenter. Matters only for -server certificates. Defaults to dc1.")
```

Comment:
We don't generate client certs in this job.
```suggestion
	c.flags.StringVar(&c.dc, "dc", "dc1", "Datacenter of the Consul cluster. Defaults to dc1.")
```